File: PASS_10/Practice_10/hash_function_example/Person.java
//  Person.java
//  Description: a Person class (Java-style)
//  Created by Glenn Lum on 1/1/20.
//  Copyright © 2020 Glenn Lum. All rights reserved.


public class Person 
{
	//a first name
	String firstName; 
	
	//a surname
	String lastName; 

	
	//Person parameterized constructor
	public Person(String firstName, String lastName) 
	{
		super();
		this.firstName = firstName;
		this.lastName = lastName;
	}


	//Displays string info of a person object
	@Override
	public String toString() 
	{
		return firstName+" "+lastName;
	}
	

	//Returns a hash code for the person object created
	@Override
	public int hashCode() 
	{
		final int prime = 31;
		int result = 1;
		result = prime * result + ((firstName == null) ? 0 : firstName.hashCode());
		result = prime * result + ((lastName == null) ? 0 : lastName.hashCode());
		return result;
	}
}

File: PASS_03/Practice_3/Person.cpp
 //  Person.cpp (v2)
 //  Description: Version 2 of the Person class with operator overloading
 //  Created by Glenn Lum on 1/1/20.
 //  Copyright © 2020 Glenn Lum. All rights reserved.

 #include "Person.h"


 Person::Person()
 {
     name="none";
     age=0;
 }
 //constructor

 Person::Person(string nm, int ag)
 {
     name=nm;
     age=ag;
 }
 //non-defsult constructor (overloading)

 Person::~Person()
 {
   //empty
 }
 //destructor


 string Person::getName() const //const =function cannot modify the class's variables
 {
     return name;
 }
 //getter

 int Person::getAge() const //const =function cannot modify the class's variables
 {
     return age;
 }
 //getter


 void Person::setName(string nm)
 {
     name = nm;
 }
 //setter

 void Person::setAge(int ag)
 {
     age = ag;
     
 }
 //setter

 void Person::operator = (const Person& P )
 {
     name=P.getName();
     age=P.getAge();
 }
 //BONUS: overloaded '=' operator !


 ostream& operator << (ostream &os, const Person &P)
 {
     os<<"Name: "<<P.getName()<<" Age: "<<P.getAge()<<endl;
     
     return os;
 }
 //overloaded << operator
 //no longer friend of the class (access data via getters and setters only)

 istream& operator >> (istream &is, Person &P)
 {
     string name;
     int age;
     
     if(is.peek()!='\n') //checks the first character in the stream for '/n'
     {
         getline(is,name,',');
         is>>age;
     }
     else
     {
         is.ignore(1); //reads from the second character onwards (why do this?)
         getline(is,name,',');
         is>>age;
     }
     
     P.setName(name);
     P.setAge(age);
     
     return is;
 }
 //overloaded >> operator
 //no longer friend of the class (access data via getters and setters only)
